internal/pkg/mapper/configmetadata/remote_test.go

```diff
@@ -87,8 +106,39 @@ func TestConfigMetadataMapAfterRemoteLoad(t *testing.T) {
 	changes := model.NewRemoteChanges()
 	changes.AddLoaded(configState)
 	changes.AddLoaded(configState2)
-
 	assert.NoError(t, mockedState.Mapper().OnRemoteChange(changes))
 	assert.Equal(t, map[string]string{"KBC.KaC.Meta": "value1", "KBC.KaC.Meta2": "value2"}, config.Metadata)
 	assert.Equal(t, make(map[string]string), config2.Metadata)
 }
+
+func TestConfigMetadataOnRemoteChangeSaved(t *testing.T) {
+	t.Parallel()
+	mockedState, httpTransport := InitState(t)
+
+	content := orderedmap.New()
+	json.MustDecodeString("{}", content)
+
+	// Config with metadata
+	configKey := model.ConfigKey{
+		BranchId:    123,
+		ComponentId: "keboola.ex-aws-s3",
+		Id:          "456",
+	}
+	configManifest := &model.ConfigManifest{
+		ConfigKey: configKey,
+	}
+	config := &model.Config{ConfigKey: configKey, Content: content, Metadata: map[string]string{"KBC.KaC.meta1": "val1"}}
+	configState := &model.ConfigState{
+		ConfigManifest: configManifest,
+		Remote:         config,
+	}
+	assert.NoError(t, mockedState.Set(configState))
+
+	// Invoke
+	changes := model.NewRemoteChanges()
+	changes.AddSaved(configState)
+	assert.NoError(t, mockedState.Mapper().OnRemoteChange(changes))
+	updateCallCount, found := httpTransport.GetCallCountInfo()["POST =~/storage/branch/123/components/keboola.ex-aws-s3/configs/456/metadata"]
+	assert.True(t, found)
+	assert.Equal(t, 1, updateCallCount)
```

Comment:
:+1: 

Stacilo to dat rovno do `Equal`
```
assert.Equal(t, 1, httpTransport.GetCallCountInfo()["POST =~/storage/branch/123/components/keboola.ex-aws-s3/configs/456/metadata"])
```

Ak sa zaznam nenajde, tak `map` vracia vzdy default hodnotu, co je v pripade cisla `0`.

Reply:
Nemusis to menit, to len poznamka bokom, ktora sa moze hodit.